com.wso2telco.dep.common.mediation/src/main/java/com/wso2telco/dep/common/mediation/spendlimit/mediator/CheckLimitMediator.java

```diff
@@ -0,0 +1,114 @@
+package com.wso2telco.dep.common.mediation.spendlimit.mediator;
+
+
+import com.wso2telco.core.dbutils.DBUtilException;
+import com.wso2telco.core.dbutils.DbService;
+import com.wso2telco.core.dbutils.dao.SpendLimitDAO;
+import com.wso2telco.dep.common.mediation.spendlimit.CustomException;
+import com.wso2telco.dep.common.mediation.spendlimit.unmarshaller.GroupDTO;
+import com.wso2telco.dep.common.mediation.spendlimit.unmarshaller.GroupEventUnmarshaller;
+import com.wso2telco.dep.common.mediation.spendlimit.unmarshaller.OperatorNotInListException;
+import org.apache.synapse.MessageContext;
+import org.apache.synapse.SynapseException;
+import org.apache.synapse.mediators.AbstractMediator;
+
+import javax.xml.bind.JAXBException;
+
+public class CheckLimitMediator extends AbstractMediator {
+
+    private static GroupEventUnmarshaller groupEventUnmarshaller;
+    private DbService dbservice = new DbService();
+
+    public static GroupEventUnmarshaller loadGroupEventUnmashaller(String xml) {
+        try {
+            GroupEventUnmarshaller.startGroupEventUnmarshaller(xml);
+        } catch (JAXBException e) {
+            return null;
+        }
+
+        return GroupEventUnmarshaller.getInstance();
+    }
+
+    public boolean mediate(MessageContext mc) throws CustomException {
```

Comment:
The entire CustomException class is not needed. In the new implementation when errorneous conditions happen it has to be handled separately.